tensorflow/lite/micro/kernels/softmax.cc

```diff
@@ -72,6 +101,68 @@ TfLiteStatus CalculateSoftmaxParams(TfLiteContext* context,
 
 }  // namespace
 
+void* SoftmaxInit(TfLiteContext* context, const char* buffer, size_t length) {
+  TFLITE_DCHECK(context->AllocatePersistentBuffer != nullptr);
+  void* raw_op_data = nullptr;
+  if (context->AllocatePersistentBuffer(context, sizeof(SoftmaxOpData),
+                                        &raw_op_data) == kTfLiteError) {
+    return nullptr;
+  }
+  SoftmaxOpData* op_data = reinterpret_cast<SoftmaxOpData*>(raw_op_data);
+  const TfLiteType input_type = context->tensors[0].type;
+  // Only allocate LUTs for KTfLiteInt16 data type
+  if (input_type == kTfLiteInt16) {
+    void* raw_exp_lut;
+    context->AllocatePersistentBuffer(context, sizeof(int16_t) * size_of_lut,
+                                      &raw_exp_lut);
+    op_data->exp_lut = reinterpret_cast<int16_t*>(raw_exp_lut);
+    void* one_over_one_plus_x_lut;
+    context->AllocatePersistentBuffer(context, sizeof(int16_t) * size_of_lut,
+                                      &one_over_one_plus_x_lut);
+    op_data->one_over_one_plus_x_lut =
+        reinterpret_cast<int16_t*>(one_over_one_plus_x_lut);
+  }
+  return raw_op_data;
+}
+
+TfLiteStatus SoftmaxPrepare(TfLiteContext* context, TfLiteNode* node) {
+  TF_LITE_ENSURE_EQ(context, NumInputs(node), 1);
+  TF_LITE_ENSURE_EQ(context, NumOutputs(node), 1);
+  const TfLiteTensor* input = GetInput(context, node, 0);
+  TF_LITE_ENSURE(context, NumDimensions(input) >= 1);
+
+  TfLiteTensor* output = GetOutput(context, node, 0);
+
+  TFLITE_DCHECK(node->user_data != nullptr);
+  SoftmaxOpData* data = new (node->user_data) SoftmaxOpData;
```

Comment:
this should be a static_cast. If you rebase then an internal change has already changed this but you may have to resolve the conflicts.

Reply:
Done.
Changed to static_cast.